<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Page</class>
 <widget class="QWidget" name="Page">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>960</width>
    <height>649</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_5">
   <property name="leftMargin">
    <number>4</number>
   </property>
   <property name="topMargin">
    <number>4</number>
   </property>
   <property name="rightMargin">
    <number>4</number>
   </property>
   <property name="bottomMargin">
    <number>4</number>
   </property>
   <item>
    <widget class="QWidget" name="widgetController" native="true">
     <layout class="QGridLayout" name="gridLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="2" column="0" colspan="2">
       <layout class="QVBoxLayout" name="verticalLayoutDeviceController"/>
      </item>
      <item row="10" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QPushButton" name="pushButtonOutputSettings">
          <property name="text">
           <string>Settings</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonOutputClear">
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="13" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Cycle interval</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QComboBox" name="comboBoxDeviceTypes"/>
      </item>
      <item row="15" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="pushButtonInputSettings">
          <property name="text">
           <string>Settings</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonInputWriteBytes">
          <property name="text">
           <string>Send</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="13" column="1">
       <widget class="QComboBox" name="comboBoxInputInterval">
        <property name="toolTip">
         <string notr="true">ms</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="1">
         <widget class="QPushButton" name="pushButtonDeviceOpen">
          <property name="text">
           <string>Open</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButtonDeviceSettings">
          <property name="text">
           <string>Settings</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="11" column="0">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="7" column="1">
       <widget class="QComboBox" name="comboBoxOutputFormat"/>
      </item>
      <item row="8" column="0" colspan="2">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QCheckBox" name="checkBoxOutputRx">
          <property name="text">
           <string>Rx</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCheckBox" name="checkBoxOutputTx">
          <property name="text">
           <string>Tx</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="checkBoxOutputDate">
          <property name="text">
           <string>Date</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QCheckBox" name="checkBoxOutputFlag">
          <property name="text">
           <string>Flag</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="checkBoxOutputTime">
          <property name="text">
           <string>Time</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QCheckBox" name="checkBoxOutputMs">
          <property name="text">
           <string>Ms</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="14" column="1">
       <widget class="QComboBox" name="comboBoxInputFormat"/>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Communication control</string>
        </property>
       </widget>
      </item>
      <item row="14" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Text format</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QLabel" name="label_2">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Output control</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QCheckBox" name="checkBoxWrap">
          <property name="text">
           <string>Wrap</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBoxTerminalMode">
          <property name="text">
           <string>Terminal</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="12" column="0" colspan="2">
       <widget class="QLabel" name="label_3">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Input control</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Text format</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabIO">
      <attribute name="title">
       <string>IO</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QWidget" name="widgetOutput" native="true">
         <layout class="QGridLayout" name="gridLayout_5">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="1" column="0">
           <widget class="QTextBrowser" name="textBrowserOutput"/>
          </item>
          <item row="1" column="1">
           <widget class="OutputPanelsManager" name="widgetOutputPanels" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <layout class="QHBoxLayout" name="horizontalLayoutOutput">
            <property name="spacing">
             <number>6</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_4">
              <property name="font">
               <font>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>Output</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayoutOutputPanelsController"/>
            </item>
            <item>
             <widget class="QWidget" name="widgetChartsController" native="true"/>
            </item>
            <item>
             <widget class="QLabel" name="label_8">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string notr="true"/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QWidget" name="widgetRxTxInfo" native="true">
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>0</height>
               </size>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLabel" name="labelRx">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">color:blue</string>
                 </property>
                 <property name="text">
                  <string notr="true">Rx</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="labelRxInfo">
                 <property name="text">
                  <string notr="true"/>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="Line" name="line_4">
                 <property name="orientation">
                  <enum>Qt::Orientation::Vertical</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="labelTx">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">color:green</string>
                 </property>
                 <property name="text">
                  <string notr="true">Tx</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="labelTxInfo">
                 <property name="text">
                  <string notr="true"/>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="toolButtonRightPanel">
              <property name="text">
               <string notr="true"/>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widgetInput" native="true">
         <layout class="QGridLayout" name="gridLayout_3">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="1" column="0">
           <widget class="QLineEdit" name="lineEditInput">
            <property name="clearButtonEnabled">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="1" rowspan="2">
           <widget class="InputPanelsManager" name="widgetInputPanels" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPlainTextEdit" name="plainTextEditInput">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_10">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="font">
               <font>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>Input</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="toolButtonInputPreset">
              <property name="toolTip">
               <string>Preset data list</string>
              </property>
              <property name="text">
               <string notr="true">ðŸ“„</string>
              </property>
              <property name="toolButtonStyle">
               <enum>Qt::ToolButtonStyle::ToolButtonTextOnly</enum>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayoutInputPanelsController"/>
            </item>
            <item>
             <widget class="QLabel" name="labelInfo">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string notr="true"/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="labelCrc">
              <property name="text">
               <string notr="true"/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="PresetView" name="tabPreset">
      <attribute name="title">
       <string>Presets</string>
      </attribute>
     </widget>
     <widget class="EmitterView" name="tabEmitter">
      <attribute name="title">
       <string>Emitter</string>
      </attribute>
     </widget>
     <widget class="ResponderView" name="tabResponder">
      <attribute name="title">
       <string>Responder</string>
      </attribute>
     </widget>
     <widget class="TransfersView" name="tabTransfers">
      <attribute name="title">
       <string>Transfers</string>
      </attribute>
     </widget>
     <widget class="ProtocolFactory" name="tabProtocolFactory">
      <attribute name="title">
       <string>Protocol Factory</string>
      </attribute>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="ScriptsManager" name="widgetScripts" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PresetView</class>
   <extends>QWidget</extends>
   <header location="global">page/preset/presetview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ResponderView</class>
   <extends>QWidget</extends>
   <header location="global">page/responder/responderview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>EmitterView</class>
   <extends>QWidget</extends>
   <header location="global">page/emitter/emitterview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TransfersView</class>
   <extends>QWidget</extends>
   <header location="global">page/transfer/transfersview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>InputPanelsManager</class>
   <extends>QWidget</extends>
   <header location="global">page/panels/inputpanels/inputpanelsmanager.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>OutputPanelsManager</class>
   <extends>QWidget</extends>
   <header location="global">page/panels/outputpanels/outputpanelsmanager.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ScriptsManager</class>
   <extends>QWidget</extends>
   <header location="global">page/scripts/scriptsmanager.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ProtocolFactory</class>
   <extends>QWidget</extends>
   <header location="global">page/protocol/protocolfactory.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>comboBoxDeviceTypes</tabstop>
  <tabstop>pushButtonDeviceSettings</tabstop>
  <tabstop>pushButtonDeviceOpen</tabstop>
  <tabstop>comboBoxOutputFormat</tabstop>
  <tabstop>checkBoxOutputRx</tabstop>
  <tabstop>checkBoxOutputTx</tabstop>
  <tabstop>checkBoxOutputFlag</tabstop>
  <tabstop>checkBoxOutputDate</tabstop>
  <tabstop>checkBoxOutputTime</tabstop>
  <tabstop>checkBoxOutputMs</tabstop>
  <tabstop>pushButtonOutputSettings</tabstop>
  <tabstop>pushButtonOutputClear</tabstop>
  <tabstop>comboBoxInputInterval</tabstop>
  <tabstop>comboBoxInputFormat</tabstop>
  <tabstop>pushButtonInputSettings</tabstop>
  <tabstop>pushButtonInputWriteBytes</tabstop>
  <tabstop>tabWidget</tabstop>
  <tabstop>textBrowserOutput</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pushButtonOutputClear</sender>
   <signal>clicked()</signal>
   <receiver>textBrowserOutput</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>292</y>
    </hint>
    <hint type="destinationlabel">
     <x>231</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
